lib/internal/event_target.js

```diff
@@ -59,7 +62,12 @@ const kBubbles = Symbol('bubbles');
 const kComposed = Symbol('composed');
 const kPropagationStopped = Symbol('propagationStopped');
 
-const isTrusted = () => false;
+const isTrustedSet = new SafeWeakSet();
+const isTrusted = ObjectGetOwnPropertyDescriptor({
+  get isTrusted() {
+    return isTrustedSet.has(this);
+  }
+}, 'isTrusted').get;
 
 class Event {
   constructor(type, options) {
```

Comment:
To ensure that the `Event` constructor has the correct `length`, `options` should be initialised to `null`:
```suggestion
  constructor(type, options = null) {
```

This also makes it possible to use strict equality comparison when checking `options !== null`.

Reply:
This sounds like a separate change from this one - @jasnell given past experience - would you prefer it if I made it here or if we opened an issue and it went on a separate PR?

Reply:
Either works I think. If you do it here make it a separate Commit but a separate pr also works

Reply:
Actually, I don't understand this - IIUC `length` is determined by the number of arguments - so `Event.length`  should be 2 anyway.

I can add a test for that though - I'll push it in https://github.com/nodejs/node/pull/35806

Reply:
Added in https://github.com/nodejs/node/pull/35806/files#diff-7177d219f295c5297f8e5d119cea177fc0b50c1df33161729ec0e1213f6c3b24R412

Reply:
`Event.length` is the number of non-optional arguments, and `options = null` makes the argument optional, thus the `length` will be `1`:

<https://tc39.es/ecma262/#sec-function-definitions-static-semantics-expectedargumentcount>

Reply:
I fixed it in the other PR :]